### ISSUE TITLE
I want take 2 box as a class,How to changle the train code?

### ISSUE THREAD

**Issue by feixuedu1106** [AlexeyAB/darknet#367]
I want take take two rectangle as a class, so I changle the cfg file, the coords as 8. and changle the region_layer.c code. but when I build the project. it so many error. my label txt like this: 0 0.246094 0.377777 0.039063 0.0777773 0.673234 0.644532 0.039063 0.0777773.
could you tell me how to changle the code like this lable for train? because in the region_layer.c file, there so many number that I did't know it's mean. thank you.

**Comment by AlexeyAB** [comment]
* What changes did you do in the region layer?
* And what error did you get?

I never did this, but **I think there isn't simple way to do this**. Because for training there is data augumentation that is hardcoded for 4-coords: 
* https://github.com/AlexeyAB/darknet/blob/993e3a38aab390625a1ac2b425c75883bbc684fe/src/data.c#L664-L714
* https://github.com/AlexeyAB/darknet/blob/993e3a38aab390625a1ac2b425c75883bbc684fe/src/data.c#L295-L331

Also you should change many lines like this:
* value `4` here: https://github.com/AlexeyAB/darknet/blob/993e3a38aab390625a1ac2b425c75883bbc684fe/src/region_layer.c#L155
* value `5` here: https://github.com/AlexeyAB/darknet/blob/993e3a38aab390625a1ac2b425c75883bbc684fe/src/region_layer.c#L172
and here: https://github.com/AlexeyAB/darknet/blob/993e3a38aab390625a1ac2b425c75883bbc684fe/src/region_layer.c#L30
and others values `5` in many other places

**Comment by feixuedu1106** [comment]
Thanks AlexeyAB. The error I mean is the train result. the detect result is't what I want. I want use darknet to detect park two corner. and set two corner as a class. do you think the YOLO v2 can do this in theory? if can do, I will try it. if not, I will quit.

**Comment by AlexeyAB** [comment]
@feixuedu1106 Does it mean "park two corner" that you want to detect just 2 points as 1 object? Why not use single default bounded box (rectangle) that already is implemented in Yolo, where top-left corner is a 1-st point and bottom-right corner is a 2-nd point.

If you want to detect 8 coords, then you can do it using Yolo: 
* but you should rewrite `load_data_detection()` and `fill_truth_detection()` to load 8 coords instead of 4 in the `data.c`
* and many places in the `region_layer.c` to train for 8 coords

**Comment by feixuedu1106** [comment]
@AlexeyAB  Hello, I edit all the read data file and in the region_layer.c 5 or 4. there code all upload . I want detect the park 3 corner. all corner use YOLO to detect  rectangle. and the center is what I need.  would you can help me check the code? thank you very much.  


![log](https://user-images.githubusercontent.com/6083782/35844853-96de3af2-0b4a-11e8-9b81-04f269093320.jpg)
![park_corner_detect](https://user-images.githubusercontent.com/6083782/35844854-9715272e-0b4a-11e8-9dd4-026b71f251a2.jpg)

**Comment by feixuedu1106** [comment]
Github could't upload code file, my email is feixuedu1106@163.com. could you have email, and I will sent the file by email to you.

**Comment by AlexeyAB** [comment]
@feixuedu1106 Try to pack your code to zip-file, and drug-n-drop here in your massage.

* So as I see did you want to detect 3 points (6 coords)?
* And is the general task the detection rotated rectangles?

**Comment by feixuedu1106** [comment]
@AlexeyAB yes, and other situation,like incline park. the image is one type in my work.
in the zip file，it have code I edit, and the config file I take coords to 12, and the train image. the label txt saved like this: id, x1, y1, w1, h1, x2, y2, w2, h2, x3, y3, w3, h3. Thank you for your help.

[code_img_cfg.zip](https://github.com/AlexeyAB/darknet/files/1701328/code_img_cfg.zip)

**Comment by feixuedu1106** [comment]
@AlexeyAB  does it can do this detect. the result I want is to detect 3 point. but YOLO just detect box. so my label file like this. If have some good advice, I will try it.

**Comment by TheMikeyR** [comment]
@feixuedu1106 can you just train it to detect it as a box and then you have, x, y, w and h as output.
You can then calculate all the three corners, see below (function made in python).
```
def main():
    # Box in yolo format
    box = [0.19646017699115, 0.0723920863309352, 0.0389380530973451, 0.0656474820143885]
    # Resolution for image
    image_size = [2208, 1242]
    box_in_pixels = convert2Pixels(image_size, box)
    red_corner, green_corner, blue_corner = convert2Corners(box_in_pixels)


def convert2Corners(box_in_pixels):
    x_min, y_min, x_max, y_max = box_in_pixels
    red_corner = x_min, y_max
    green_corner = x_max, y_max
    blue_corner = x_max, y_min
    return red_corner, green_corner, blue_corner


def convert2Pixels(image_size, box):
    height, width = image_size[1], image_size[0]
    x_min = width * float(box[0])
    y_min = height * float(box[1])
    w = width * float(box[2])
    h = height * float(box[3])
    x_min = x_min - (w / 2)
    y_min = y_min - (h / 2)
    x_max = x_min + w
    y_max = y_min + h
    return (x_min, y_min, x_max, y_max)


if __name__ == "__main__":
    main()

```

**Comment by feixuedu1106** [comment]
@TheMikeyR ， now I use this way. 3 points is not just I want , the 3 point compose as park is what I really want. I want 3 point as a park. but thank you for your reply.
